executor/executor.go

```diff
@@ -735,6 +735,10 @@ func (e *CheckTableExec) handlePanic(r interface{}) {
 
 // Next implements the Executor Next interface.
 func (e *CheckTableExec) Next(ctx context.Context, req *chunk.Chunk) error {
+	tempTableType := e.table.Meta().TempTableType
+	if tempTableType == model.TempTableGlobal || tempTableType == model.TempTableLocal {
+		return errors.Trace(ErrAdminCheckTable)
```

Comment:
Can you check this earlier? It's better to be checked in preprocessing or optimizer phase. If you can't, it's better to be in building the executor.